src/components/page_template/page_template.tsx

```diff
@@ -0,0 +1,216 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import React, { FunctionComponent, useEffect, useState } from 'react';
+import classNames from 'classnames';
+
+import { _EuiPageOuter as EuiPageOuter, _EuiPageOuterProps } from './outer';
+import { _EuiPageInner as EuiPageInner, _EuiPageInnerProps } from './inner';
+import { _EuiPageSidebar as EuiPageSidebar } from './sidebar';
+import { _EuiPageBottomBar as EuiPageBottomBar } from './bottom_bar/page_bottom_bar';
+import { _EuiPageEmptyPrompt as EuiPageEmptyPrompt } from './empty_prompt/page_empty_prompt';
+import { EuiPageHeader, EuiPageSection } from '../page';
+import { _EuiPageRestrictWidth } from '../page/_restrict_width';
+import { _EuiPageBottomBorder } from '../page/_bottom_border';
+import { useEuiTheme } from '../../services';
+import { logicalStyle, logicalUnit } from '../../global_styling';
+
+export type EuiPageTemplateProps = _EuiPageOuterProps &
+  // We re-define the `border` prop below to be named more appropriately
+  Omit<_EuiPageInnerProps, 'border'> &
+  _EuiPageRestrictWidth &
+  _EuiPageBottomBorder & {
+    /**
+     * Applies a top or left border to the inner contents
+     * to add separation between content and sidebar when a sidebar exists.
+     */
+    contentBorder?: _EuiPageInnerProps['border'];
+    /**
+     * Minimum height in which to enforce scrolling
+     */
+    minHeight?: string;
+    /**
+     * To account for any fixed elements like headers,
+     * pass in the value of the total height of those fixed elements.
+     * Otherwise they will be calculated based on the data attributes on the body element.
+     */
+    offset?: number;
+  };
+
+/**
+ * Consumed via `EuiPageTemplate`,
+ * it controls and propogates most of the shared props per direct child
+ */
+export const _EuiPageTemplate: FunctionComponent<EuiPageTemplateProps> = ({
+  children,
+  // Shared props
+  restrictWidth = true,
+  paddingSize = 'l',
+  grow = true,
+  bottomBorder,
+  offset: _offset,
+  panelled,
+  // Inner props
+  contentBorder,
+  // Outer props
+  className,
+  minHeight = '460px',
+  responsive = ['xs', 's'],
+  ...rest
+}) => {
+  const { euiTheme } = useEuiTheme();
+
+  const [offset, setOffset] = useState(_offset);
+
+  useEffect(() => {
+    if (_offset === undefined) {
+      const euiHeaderFixedCounter = document.body.getAttribute(
+        'data-fixed-headers'
+      );
+
+      setOffset(euiTheme.base * 3 * Number(euiHeaderFixedCounter));
```

Comment:
```suggestion
      const euiHeaderFixedCounter = Number(document.body.dataset.fixedHeaders);
      setOffset(euiTheme.base * 3 * euiHeaderFixedCounter);
```

minor syntax suggestion, `dataset` is a more modern way of grabbing data attrs than `getAttribute`